webui/react/src/components/ExperimentRetainLogsModal.tsx

```diff
@@ -0,0 +1,148 @@
+
+const FORM_ID = 'retain-experiment-logs-form';
+const FOREVER = -1;
+
+type FormInputs = {
+  numDays: number;
+};
+
+interface Props {
+  excludedExperimentIds?: Set<number>;
+  experimentIds: number[];
+  filters?: V1BulkExperimentFilters;
+  onSubmit?: (successfulIds?: number[]) => void;
+}
+
+const ExperimentRetainLogsModalComponent: React.FC<Props> = ({
+  excludedExperimentIds,
+  experimentIds,
+  filters,
+  onSubmit,
+}: Props) => {
+  const idPrefix = useId();
+  const { openToast } = useToast();
+  const [checked, setChecked] = useState<boolean>(false);
+  const [form] = Form.useForm<FormInputs>();
+  const inputDays = Form.useWatch('numDays', form);
+
+  const handleCheckBoxChange = useCallback(
+    (event: CheckboxChangeEvent) => {
+      const isChecked = event.target.checked;
+      setChecked(isChecked);
+      if (isChecked) {
+        form.setFieldValue('numDays', FOREVER);
+      } else {
+        form.setFieldValue('numDays', null);
+      }
+    },
+    [form],
+  );
+
+  const handleSubmit = useCallback(async () => {
+    const values = await form.validateFields();
+    const numberDays = values.numDays;
+    let filt = filters;
+    if (excludedExperimentIds && excludedExperimentIds.size > 0) {
+      filt = { ...filters, excludedExperimentIds: Array.from(excludedExperimentIds) };
+    }
+    try {
+      const results = await changeExperimentLogRetention({
+        experimentIds,
+        filters: filt,
+        numDays: numberDays,
+      });
+
+      onSubmit?.(results.successful);
+
+      const numSuccesses = results.successful.length;
+      const numFailures = results.failed.length;
+
+      const stringDays = numberDays === -1 ? 'forever' : `for ${numberDays} days`;
+
+      if (numFailures === 0) {
+        openToast({
+          closeable: true,
+          description: `Retained logs for ${results.successful.length} ${pluralizer(
+            numSuccesses,
+            'experiment',
+          )} ${stringDays}`,
+          title: 'Retain Logs Success',
+        });
+      } else if (numSuccesses === 0) {
+        openToast({
+          description: `Unable to retain logs for ${numFailures} ${pluralizer(
+            numFailures,
+            'experiment',
+          )}`,
+          severity: 'Warning',
+          title: 'Retain Logs Failure',
+        });
+      } else {
+        openToast({
+          closeable: true,
+          description: `Failed to retain logs for ${numFailures} experiments out of ${
+            numFailures + numSuccesses
+          } for ${numberDays} days.`,
+          severity: 'Warning',
+          title: 'Partial Retain Logs Failure',
+        });
+      }
+    } catch (e) {
+      handleError(e, { publicSubject: 'Unable to retain logs' });
+    }
+  }, [form, excludedExperimentIds, experimentIds, filters, onSubmit, openToast]);
+
+  return (
+    <Modal
+      cancel
+      size="small"
+      submit={{
+        disabled: inputDays == null,
```

Comment:
good to have: even though input is an invalid value, submit button is not disabled. good to be disabled with an invalid value as well empty value.

<img width="414" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/048b8337-ec63-462d-8156-0a98ccb7cd55">

Reply:
fixed